### PR TITLE
Add support for RJ45 speeds on dynamic breakout mode

### PR DESCRIPTION
**Pull request by tomer-israel**
#### Why I did it
need to support the new speeds 1G & 100M on the breakout modes

#### How I did it
added to portconfig.py the support for speeds with character 'M' (Mega)

#### How to verify it
configure manually speeds on hwsku & platform.json the breakout mode 1G[100M] & 100M[1G]

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [x] 202106

#### Description for the changelog

Add new support for 100M speeds on breakout mode

### CODE REVIEW

File: src/sonic-yang-models/yang-models/sonic-breakout_cfg.yang [hunk 1]
```diff
@@ -47,7 +47,7 @@ module sonic-breakout_cfg {
                          * Add any other breakout-mode to allow Dynamic Port
                          * Breakout to that breakout-mode.
                          */
-                        pattern '1x100G\[40G\]|2x50G|4x25G\[10G\]|2x25G\(2\)\+1x50G\(2\)|1x50G\(2\)\+2x25G\(2\)|1x400G|2x200G|4x100G|8x50G|1x200G\[100G,50G,40G,25G,10G,1G\]|2x100G\[50G,40G,25G,10G,1G\]|4x50G\[40G,25G,10G,1G\]|1x25G\[10G\]|1x100G\[50G,40G,25G,10G\]|2x50G\[40G,25G,10G\]|4x25G\[10G\]|1x25G\[10G,1G\]|1x100G\[50G,40G,25G,10G,1G\]|2x50G\[40G,25G,10G,1G\]|4x25G\[10G,1G\]|1x400G\[200G,100G,50G,40G,25G,10G,1G\]|2x200G\[100G,50G,40G,25G,10G,1G\]|4x100G\[50G,40G,25G,10G,1G\]';
+                        pattern '1x100G\[40G\]|2x50G|4x25G\[10G\]|2x25G\(2\)\+1x50G\(2\)|1x50G\(2\)\+2x25G\(2\)|1x400G|2x200G|4x100G|8x50G|1x200G\[100G,50G,40G,25G,10G,1G\]|2x100G\[50G,40G,25G,10G,1G\]|4x50G\[40G,25G,10G,1G\]|1x25G\[10G\]|1x100G\[50G,40G,25G,10G\]|2x50G\[40G,25G,10G\]|4x25G\[10G\]|1x25G\[10G,1G\]|1x100G\[50G,40G,25G,10G,1G\]|2x50G\[40G,25G,10G,1G\]|4x25G\[10G,1G\]|1x400G\[200G,100G,50G,40G,25G,10G,1G\]|2x200G\[100G,50G,40G,25G,10G,1G\]|4x100G\[50G,40G,25G,10G,1G\]|1x1G[100M]';
```
Comment (alexrallen) [inline src/sonic-yang-models/yang-models/sonic-breakout_cfg.yang:50]: I'm not sure this change is sufficient for what you may want. This is about to be a moot point regardless when https://github.com/Azure/sonic-buildimage/pull/6801 is merged (which is hopefully imminent) but just as an FYI what you have added will ONLY allow the 1x1G[100M] mode. I.e. 100M would not be allowed in any breakouts or in any higher speed modes i.e. 1x10G[1G, 100M].

**Comment by alexrallen** [comment]
Looking at this change, it seems that the only effective change is allowing the "M" to be provided after the speed. 

By default, if "G" is not specified the speed is interpreted in Mbps. I want to clarify that is is your intention and that you are not under the impression than anything else is being functionally changed here. 

If this is intended to be upstreamed to vanilla SONiC we might run into some protest from the community for this change since it is not "functional"

**Review by alexrallen — APPROVED** [review]
Aside from my two comments which are FYIs / warnings this change looks good.

File: src/sonic-yang-models/yang-models/sonic-breakout_cfg.yang (hunk 1 above)
Comment (dprital) [inline src/sonic-yang-models/yang-models/sonic-breakout_cfg.yang:50]: It seems that you need to add  "\" to the end of definition "1G[100M\"] + add also "1G[]"

**Pull request closed without merging**